### PR TITLE
refactor: streamline retryFetch and convert most usage to retry

### PR DESCRIPTION
**Pull request by idoshamun**
I refactor our `retryFetch` function to support more generic use cases so we'll be able to apply system-wide. It's best practice to use retries and now that we know our retry library don't introduce issues we can actually apply it :P

### CODE REVIEW

File: src/integrations/retry.ts [hunk 1]
```diff
@@ -66,23 +81,29 @@ export async function asyncRetry<T>(
...
     if (res.status < 500) {
-      throw new AbortError(`request is invalid: ${res.status}`);
+      throw new AbortError(err);
     }
-    throw new Error(`unexpecetd response from feed service: ${res.status}`);
+    throw err;
   }, retryOpts);
 }
+
+export async function retryFetchParse<T>(
```
Comment (rebelchris) [inline src/integrations/retry.ts:102]: I'd only argue this placement inside `feed` dir is a bit off.

File: src/kratos.ts [hunk 2]
```diff
@@ -33,28 +33,26 @@ const fetchKratos = async (
...
-      });
-      if (res.status < 300) {
-        return { res: await res.json(), headers: res.headers };
-      }
-      const err = new KratosError(res.status, await res.text());
-      if (res.status >= 500) {
-        req.log.warn({ err }, 'unexpected error from kratos');
-        throw err;
+  try {
+    const res = await retryFetch(endpoint, {
```
Comment (rebelchris) [inline src/kratos.ts:37]: Why don't we use `retryFetchParse` here as well?

**Review by rebelchris — APPROVED** [review]
Nice, thanks for that, have some minor non blocking suggestions

File: src/kratos.ts (hunk 2 above)
Comment (idoshamun) [reply src/kratos.ts:37]: Because we need the response headers. Parse returns only the body parsed as json

File: src/integrations/retry.ts (hunk 1 above)
Comment (idoshamun) [reply src/integrations/retry.ts:102]: It's not in the `feed` dir, it's under integrations, or am I missing something?

File: src/kratos.ts (hunk 2 above)
Comment (rebelchris) [reply src/kratos.ts:37]: Ah I see :) 
Missed the headers specifically as I saw you can manually do custom error status on another call.

**Pull request merged**